Wikipedia/View Controllers/WebView/WebViewController.m

```diff
@@ -1488,17 +1490,8 @@ - (BOOL)isCurrentArticleSaved {
 #pragma mark Scroll to last section after rotate
 
 - (void)willRotateToInterfaceOrientation:(UIInterfaceOrientation)toInterfaceOrientation duration:(NSTimeInterval)duration {
-    NSString* js = @"(function() {"
-                   @"    _topElement = document.elementFromPoint( window.innerWidth / 2, 0 );"
-                   @"    if (_topElement) {"
-                   @"        var rect = _topElement.getBoundingClientRect();"
-                   @"        return rect.top / rect.height;"
-                   @"    } else {"
-                   @"        return 0;"
-                   @"    }"
-                   @"})()";
-    float relativeScrollOffset = [[self.webView stringByEvaluatingJavaScriptFromString:js] floatValue];
-    self.relativeScrollOffsetBeforeRotate = relativeScrollOffset;
+    [[self.webView wmf_javascriptContext][@"setPreRotationRelativeScrollOffset"] callWithArguments:nil];
```

Comment:
This whole way of dealing with rotation is a bit hacky, but at least this contains the hackiness mostly in the JS file for now.